components/brave_new_tab_ui/containers/newTab/settings/braveToday/publisherPrefs.tsx

```diff
@@ -6,24 +6,48 @@
 import * as React from 'react'
 import AutoSizer from '@brave/react-virtualized-auto-sizer'
 import { VariableSizeList } from 'react-window'
-import { Publisher, UserEnabled } from '../../../../api/brave_news'
+import getBraveNewsController, { Channels, Publisher, UserEnabled } from '../../../../api/brave_news'
 import {
   SettingsRow,
   SettingsText
 } from '../../../../components/default'
 import { Toggle } from '../../../../components/toggle'
 import * as s from './style'
+import usePromise from '../../../../hooks/usePromise'
+import { defaultState as newTabData } from '../../../../storage/new_tab_storage'
+
+/**
+ * Determines whether a publishers content is shown in the feed. This might mean
+ * it belongs to an active channel, or that the user has explicitly turned it on
+ * or off, or that the feed is enabled by default (if channels are not available).
+ * @param publisher The publisher to check.
+ * @param channels All the current channels, with up to date subscriptions
+ * @returns Whether the publisher is current enabled.
+ */
+function isPublisherContentAllowed (publisher: Publisher, channels: Channels): boolean {
+  // If the user has an explicit preference for this feed, use that.
+  if (publisher.userEnabledStatus === UserEnabled.ENABLED) return true
+  if (publisher.userEnabledStatus === UserEnabled.DISABLED) return false
+
+  // If we're using the old API, check if the source is default enabled.
+  if (newTabData.featureFlagBraveNewsV2Enabled) return publisher.isEnabled
```

Comment:
If `newTabData.featureFlagBraveNewsV2Enabled` is true, I assume we're using channels API.
And maybe it's new API, then `If (!newTabData.featureFlagBraveNewsV2Enabled)` is right condition check?

Reply:
Oh good catch! Actually fixed the other check but not this one :laughing: